addons/payment_razorpay/models/payment_provider.py

```diff
@@ -129,3 +187,282 @@ def _get_validation_amount(self):
             return res
 
         return 1.0
+
+    # === BASE METHODS ===#
+
+    @api.onchange('state')
+    def _onchange_state(self):
+        for provider in self.filtered(lambda p: p.code == 'razorpay'):
+            change_from_test = provider._origin.state == 'test' and (provider.state == 'disabled' or provider.state == 'enabled')
+            if self.razorpay_public_token and (provider.state == 'test' or change_from_test):
+                self.write({
+                    'razorpay_access_token': False,
+                    'razorpay_refresh_token': False,
+                    'razorpay_access_token_expiration': False,
+                    'razorpay_public_token': False,
+                    'razorpay_key_id': False,
+                    'razorpay_key_secret': False,
+                    'razorpay_webhook_secret': False,
+                })
+
+    def _get_razorpay_oauth_url(self):
+        """
+        Get the OAuth URL for Razorpay.
+
+        :return: The Razorpay OAuth URL.
+        :rtype: str
+        """
+        self.ensure_one()
+        oauth_url = 'OAUTH_LIVE_URL'
+        if self.state == 'test':
+            oauth_url = 'OAUTH_TEST_URL'
+        return self.env['ir.config_parameter'].get_param(oauth_url)
+
+    # === ONBOARDING APIS ===#
+
+    def action_razorpay_connect_account(self):
+        """
+        Create a Razorpay Connect account and redirect the user.
+        If the provider is already enabled, close the current window. Otherwise
+        generate a Razorpay Connect onboarding link and redirect the user to it.
+        If the link generation fails, redirect the user to the provider form.
+
+        :return: URL for the next step.
+        :rtype: str
+        """
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'url': self._get_razorpay_authorize_url(),
+            'target': 'self',
+        }
+
+    def _razorpay_generate_state(self):
+        """
+        Generate a random 80-character string for use as a secure state parameter.
+
+        :return: Random string.
+        :rtype: str
+        """
+        return ''.join(secrets.choice(string.ascii_letters + string.digits) for _ in range(80))
```

Comment:
This does not seems unique for all the time :thinking: 
you can refer to `generate_idempotency_key` method to create unique state use dbuuid and id of razorpay provider